### ISSUE TITLE
Inheritance in simple usertypes doesn't work

### ISSUE THREAD

**Issue by jkbz64** [ThePhD/sol2#214]
There's problem with inheritance( or lack information about it in documentation) in simple_usertypes.
When using usertypes everything works fine, but when i'm using simple_usertypes, the program throws exception. If inheritance doesn't work with simple usertypes, then we need to add some warning/notice about it in inheritance section, because I couldn't find anything about it in documentation.
Reproduction code:
```
class A
{
public:
    virtual void a() = 0;
};

class B : public A
{
public:
    virtual void a() override { }
};

int main()
{
    sol::state state;
    state.new_simple_usertype<A>("A", "a", &A::a);
    state.new_simple_usertype<B>("B", sol::base_classes, sol::bases<A>());
    B* b = new B();
    state.set("b", b);
    state.script("b:a()");
}
```

**Comment by ThePhD** [comment]
From the docs, at the bottom of the [inheritance](http://sol2.readthedocs.io/en/latest/api/usertype.html#inheritance) portion of the usertype documentation:

> Sol currently attempts to link base class methods and variables with their derived classes with an **undocumented, unsupported feature**

Emphasis mine, but maybe I should have an explicit separate note saying how unsupported and undocumented it really is. As it stands, the 'feature' is there because people continually expect it to be: I don't endorse it because the current way of using it has performance implications and the implementation itself is somewhat messy.

Also from the docs, in the same yellow exclamation [ ! ] warning bubble:
> please **specify all the base methods** on the derived class in the method listing you write

I will also add a warning note to simple_usertype to make it more apparent how unsupported it is.

**Comment by ThePhD** [comment]
It is now explicit [here, about inheritance](http://sol2.readthedocs.io/en/latest/api/usertype.html#inheritance) (in the warning) and [here, for simple_usertype](http://sol2.readthedocs.io/en/latest/api/simple_usertype.html).

**Comment by ThePhD** [comment]
So this bothered me a lot. In the end, I actually ended up working hard on the implementation and enabling inheritance, as well as actual properties / variables. The usual caveats no longer apply: http://sol2.readthedocs.io/en/latest/api/simple_usertype.html

Give it a try with the latest (v2.14.0), and let me know if it's satisfactory.

**Comment by jkbz64** [comment]
Everything works perfectly, thanks for the new update. Keep up the good work!

**Comment by DannyDaiSun** [comment]
@ThePhD Hi, first of all thanks a lot for this great library, so useful. Second, regarding to your message: 

> So this bothered me a lot. In the end, I actually ended up working hard on the implementation and enabling inheritance, as well as actual properties / variables. The usual caveats no longer apply: http://sol2.readthedocs.io/en/latest/api/simple_usertype.html
> 
> Give it a try with the latest (v2.14.0), and let me know if it's satisfactory.

does it mean, we no longer need to specify base class member methods when registering derived class via new_usertype<T>? I have tried the following, which works as I expected, but I just worried if there would be a runtime overhead issue?

my code:

    new_usertype<widget>(
        "widget",
        "draw", &widget::draw,
        );

    new_usertype<button>(
        "button",
        "draw", &button::draw,
        sol::base_classes, sol::bases<widget>()
        );

where "button" derived from "widget" with "draw" being the overloaded member function.

In summary, I wonder if my code is the best way of doing inheritance with sol?

Thank you in advance.
